### PR TITLE
Extend voting message

### PR DESCRIPTION
**Pull request by mstrasinskis**
# Motivation

- added more detailed voting status message (1/3 -> 2/3 -> 3/3 -> updating)
- because the voting logic became quite big, it was moved from `proposal.services` to `vote-registration.services` (same for `vote-registration.store`)

# Changes

- voting status message displays more info (discussed w/ Mischa)
- error handling was updated to catch more

## Refactoring
- `voting.store` -> `vote-registration.store`
-  vote related part from `proposal.services` -> `vote-registration.services`

# Tests

- voting related tests were moved to `vote-registration.services.ts`
- `vote-registration.store.spec`
- added register vote status message tests

# Screenshots

https://user-images.githubusercontent.com/98811342/188685991-b5a0b312-affb-4956-8412-fd0fb7dccbfe.mp4

### CODE REVIEW

File: frontend/src/lib/services/vote-registration.services.ts [hunk 1]
```diff
@@ -0,0 +1,375 @@
...
+  vote,
+  reloadProposalCallback: updateProposalContext,
+}: {
+  neuronIds: NeuronId[];
+  proposalInfo: ProposalInfo;
+  vote: Vote;
+  reloadProposalCallback: (proposalInfo: ProposalInfo) => void;
+}): Promise<void> => {
+  try {
+    const toastId = createRegisterVotesToast({ vote, proposalInfo, neuronIds });
```
Comment (peterpeterparker) [inline frontend/src/lib/services/vote-registration.services.ts:49]: maybe rename `createRegisterVotesToast` in `initRegisterVotesToast`?

Feels a bit weird at reading to call `createRegisterVotesToast` and then another `voteRegistrationStore.create` ("create create")

File: frontend/src/lib/services/vote-registration.services.ts [hunk 2]
```diff
@@ -0,0 +1,375 @@
...
+    const proposalId = proposalInfo.id as bigint;
+    voteRegistrationStore.create({
```
Comment (peterpeterparker) [inline frontend/src/lib/services/vote-registration.services.ts:51]: why should `proposalInfo, neuronIds` be passed again, in previous method they were provided. Can we spare this?

File: frontend/src/lib/services/vote-registration.services.ts [hunk 3]
```diff
@@ -0,0 +1,375 @@
...
+      vote,
+      proposalInfo,
+      neuronIds,
+      toastId,
+    });
+
+    voteRegistrationStore.updateStatus({
```
Comment (peterpeterparker) [inline frontend/src/lib/services/vote-registration.services.ts:58]: any particular reason why we don't use the `toastId` here?

a bit confused why we have a toast id and a proposal id, feels like both are unique no?

File: frontend/src/lib/services/vote-registration.services.ts [hunk 4]
```diff
@@ -0,0 +1,375 @@
...
+      proposalId,
+      status: "complete",
+    });
+
+    toastsStore.hide(toastId);
+
+    voteRegistrationStore.removeCompleted();
+  } catch (err: unknown) {
+    console.error("vote unknown:", err);
```
Comment (peterpeterparker) [inline frontend/src/lib/services/vote-registration.services.ts:101]: should we clean the `voteRegistrationStore` of previous msg on error?

File: frontend/src/lib/services/vote-registration.services.ts [hunk 5]
```diff
@@ -0,0 +1,375 @@
...
+      neuronIds,
+      toastId,
+    });
+
+    voteRegistrationStore.updateStatus({
+      proposalId,
+      status: "vote-registration",
+    });
+
+    await registerNeuronsVote({
```
Comment (peterpeterparker) [inline frontend/src/lib/services/vote-registration.services.ts:63]: can you add some comments like

```
1. we are doing this
2. now we are doing this
```

it's quite a long function would help to follow

File: frontend/src/lib/services/vote-registration.services.ts [hunk 6]
```diff
@@ -0,0 +1,375 @@
...
+  const completeNeurons = successfullyVotedNeuronIds.length;
+  const status = registrationDone
+    ? $i18n.proposal_detail__vote.vote_status_updating
+    : replacePlaceholders($i18n.proposal_detail__vote.vote_status_registering, {
+        $completed: `${completeNeurons}`,
+        $amount: `${totalNeurons}`,
+      });
+
+  toastsStore.updateToastContent({
+    toastId: toastId as symbol,
```
Comment (peterpeterparker) [inline frontend/src/lib/services/vote-registration.services.ts:238]: out of curiosity, why the cast?

File: frontend/src/lib/services/vote-registration.services.ts [hunk 7]
```diff
@@ -0,0 +1,375 @@
...
+  reloadProposalCallback: (proposalInfo: ProposalInfo) => void;
+}): Promise<void> => {
+  try {
+    const toastId = createRegisterVotesToast({ vote, proposalInfo, neuronIds });
+    const proposalId = proposalInfo.id as bigint;
+    voteRegistrationStore.create({
+      vote,
+      proposalInfo,
+      neuronIds,
+      toastId,
```
[the remark this answers is not in the document]
Comment (mstrasinskis) [reply frontend/src/lib/services/vote-registration.services.ts]: was removed

File: frontend/src/lib/services/vote-registration.services.ts (hunk 5 above)
Comment (peterpeterparker) [reply frontend/src/lib/services/vote-registration.services.ts:63]: thanks

**Review by peterpeterparker — APPROVED** [review]
LGTM, thanks

(review executed together in a call)

**Pull request merged**